File: Backend/user-service/spring_boot_backend_template/spring_boot_backend_template/src/main/java/com/backend/entity/BoardingPoint.java
package com.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalTime;

@Entity
@Table(name = "boarding_points")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class BoardingPoint {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String locationName;   // Wakad, Hinjewadi, Shivajinagar

    @Column(nullable = false)
    private LocalTime boardingTime;

    @ManyToOne
    @JoinColumn(name = "schedule_id", nullable = false)
    private BusSchedule busSchedule;
}

File: Backend/admin-service/admin/admin-service/Models/BoardingPoint.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace admin_service.Models
{
    [Table("boarding_points")]
    public class BoardingPoint
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        [Column("location_name")]
        public string LocationName { get; set; }   // Wakad, Hinjewadi, Shivajinagar

        [Required]
        [Column("boarding_time")]
        public TimeSpan BoardingTime { get; set; } // LocalTime in Java maps to TimeSpan in C#

        // Foreign key property
        [Required]
        [Column("schedule_id")]
        public long BusScheduleId { get; set; }

        // Navigation property
        [ForeignKey("BusScheduleId")]
        public BusSchedule BusSchedule { get; set; }
    }
}
